dask_ml/_partial.py

```diff
@@ -213,13 +218,21 @@ def predict(model, x):
 
     See docstring for ``da.learn.fit``
     """
+    if not hasattr(x, "chunks") and hasattr(x, "to_dask_array"):
+        index = x.index
+        x = x.to_dask_array()
+    else:
+        index = None
     assert x.ndim == 2
     if len(x.chunks[1]) > 1:
         x = x.rechunk(chunks=(x.chunks[0], sum(x.chunks[1])))
     func = partial(_predict, model)
     xx = np.zeros((1, x.shape[1]), dtype=x.dtype)
     dt = model.predict(xx).dtype
-    return x.map_blocks(func, chunks=(x.chunks[0], (1,)), dtype=dt).squeeze()
+    result = x.map_blocks(func, chunks=(x.chunks[0], (1,)), dtype=dt).squeeze()
+    if index is not None:
+        result = dd.from_dask_array(result, index=index)
```

Comment:
Hmm I'm having second thoughts about this. I think this is the wrong place to determine whether a Series should be returned. I think that should be up to `model.predict`. And given the current implementation, `model.predict` is only going to return arrays, since it's only ever getting arrays.

So for now I'd recommend reverting the `predict` related changes, and just keep this to support dataframes in `fit`

Reply:
My main priority is matching the behavior of the underlying estimator. Ideally `Incremental(estimator)` is behaves as close as possible to

```
for block in array:
    estimator.partial_fit(block)
```

And prediction is a simple `map(predict, blocks)`.

So if `estimator.predict(pd.DataFrame) -> ndarray`, we would want `estimator.predict(dd.DataFrame) -> da.Array`. At the moment, IIUC, that would return a `Series`.

---

And now I realize I should have read your whole comment. Let's just delete `predict`?

Regardless that can be done elsewhere.